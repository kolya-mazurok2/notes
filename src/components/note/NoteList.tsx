import { Grid } from '@mui/material';

import { notes } from '../../data/notes';
import NoteListItem from './NoteListItem';

const NoteList = () => {
  return (
    <Grid container spacing={3}>
      {notes.map((note) => {
        return (
          <Grid key={note.id} item xs={4}>
            <NoteListItem note={note} />
          </Grid>
        );
      })}
    </Grid>
  );
};

export default NoteList;
