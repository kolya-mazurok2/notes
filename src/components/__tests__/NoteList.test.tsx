import renderer from 'react-test-renderer';
import NoteList from '../note/NoteList';

it('should render an empty message when no notes', () => {
  const element = renderer
    .create(<NoteList notes={[]} onNoteDelete={() => {}} onNoteEdit={() => {}} />)
    .toJSON();

  expect(element).toMatchSnapshot();
});

it('should render multiple notes', () => {
  const notes = [
    {
      id: 'a1',
      title: 'First',
      description: 'Lorem ipsum',
      createdAt: new Date(1655350750000)
    },
    {
      id: 'a2',
      title: 'Second',
      description: '',
      createdAt: new Date(1655350750000)
    }
  ];

  const element = renderer
    .create(<NoteList notes={notes} onNoteDelete={() => {}} onNoteEdit={() => {}} />)
    .toJSON();

  expect(element).toMatchSnapshot();
});
