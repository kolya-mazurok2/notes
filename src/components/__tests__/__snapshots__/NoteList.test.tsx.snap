// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an empty message when no notes 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      No notes have been added yet!
    </h6>
  </div>
</div>
`;

exports[`should render multiple notes 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              First
            </span>
          </div>
        </div>
        <p
          className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Lorem ipsum
        </p>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-alignRight css-h2l1uj-MuiTypography-root"
        >
          2022-6-16 6:39:10
        </h6>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Edit
          </button>
          <button
            className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-6ekzlo-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Second
            </span>
          </div>
        </div>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-alignRight css-h2l1uj-MuiTypography-root"
        >
          2022-6-16 6:39:10
        </h6>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Edit
          </button>
          <button
            className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-6ekzlo-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
